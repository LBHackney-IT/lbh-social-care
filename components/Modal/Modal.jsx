import PropTypes from 'prop-types';
import ReactModal from 'react-modal';

import style from './Modal.module.scss';

ReactModal.setAppElement('#root');

const Modal = ({ isOpen, onRequestClose, children }) => (
  <ReactModal
    isOpen={isOpen}
    onRequestClose={onRequestClose}
    className={style.modal}
    overlayClassName={style.backdrop}
    shouldCloseOnOverlayClick
  >
    <button className={style.closeButton} onClick={onRequestClose}>
      X
    </button>
    <div>{children}</div>
  </ReactModal>
);

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onRequestClose: PropTypes.func.isRequired,
};

export default Modal;
