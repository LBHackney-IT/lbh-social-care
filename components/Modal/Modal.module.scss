.modal {
  position: absolute;
  width: 90%;
  max-width: 960px;
  margin: 0;
  background-color: #fff;
  border-radius: 0.3rem;
  box-shadow: 0 0 2.5rem rgba(0, 0, 0, 0.5);
  top: 50%;
  left: 50%;
  text-align: left;
  transform: translate(-50%, -50%);
  & > div {
    padding: 2.5rem;
    overflow: auto;
    max-height: 90vh;
  }
}

.backdrop {
  background-color: rgba(0, 0, 0, 0.7);
  top: 0;
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1000;
}

.closeButton {
  position: absolute;
  top: 1rem;
  right: 1rem;
  border: none;
  padding: 0;
  background: none;
}
