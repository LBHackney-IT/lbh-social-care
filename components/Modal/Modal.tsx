import { useEffect } from 'react';
import ReactModal from 'react-modal';

import TimesCircleIcon from 'components/Icons/TimesCircle';

import style from './Modal.module.scss';

ReactModal.setAppElement('#root');

interface Props {
  isOpen: boolean;
  onRequestClose: () => void;
  children: React.ReactChild;
}

const Modal = ({
  isOpen,
  onRequestClose,
  children,
}: Props): React.ReactElement => {
  useEffect(() => {
    document.documentElement.style.overflow = isOpen ? 'hidden' : '';
    return () => {
      document.documentElement.style.overflow = '';
    };
  }, [isOpen]);
  return (
    <ReactModal
      isOpen={isOpen}
      onRequestClose={onRequestClose}
      className={style.modal}
      overlayClassName={style.backdrop}
      shouldCloseOnOverlayClick
    >
      <button className={style.closeButton} onClick={onRequestClose}>
        <TimesCircleIcon color="border" />
      </button>
      <div>{children}</div>
    </ReactModal>
  );
};

export default Modal;
