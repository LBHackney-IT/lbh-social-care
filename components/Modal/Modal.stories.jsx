import { useState } from 'react';

import Modal from './Modal';

export default {
  title: 'Modal',
  component: Modal,
};

export const Default = () => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <>
      <button onClick={() => setIsOpen(true)}>Show the Modal</button>
      <Modal isOpen={isOpen} onRequestClose={() => setIsOpen(false)}>
        I am the content of the Modal
      </Modal>
    </>
  );
};
