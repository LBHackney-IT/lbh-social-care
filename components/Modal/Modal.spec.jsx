import { render, fireEvent } from '@testing-library/react';

import Modal from './Modal';

describe(`Modal`, () => {
  const props = {
    onRequestClose: jest.fn(),
    isOpen: true,
    children: <p>I am the content!</p>,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render properly', () => {
    const { asFragment } = render(<Modal {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onRequestClose', () => {
    const { getByRole } = render(<Modal {...props} />);
    fireEvent.click(getByRole('button'));
    expect(props.onRequestClose).toHaveBeenCalled();
  });
});
