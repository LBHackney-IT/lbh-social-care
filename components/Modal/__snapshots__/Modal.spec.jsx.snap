// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render properly 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <button
      class="closeButton"
    >
      X
    </button>
    <div>
      <p>
        I am the content!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
